modules/Attendance/attendance.php

```diff
@@ -16,488 +16,448 @@
-		                                    	$class = 'highlightAbsent';
-		                                    } else {
-		                                    	$class = 'highlightNoData';
-		                                    	$link = '';
-		                                    }
-		                                }
-
-		                                echo "<td class='$class' style='padding: 12px !important;'>";
-		                                if ($link != '') {
-		                                    echo "<a href='$link'>";
-		                                    echo date('d', $currentDayTimestamp).'<br/>';
-		                                    echo "<span>".date('M', $currentDayTimestamp).'</span>';
-		                                    echo '</a>';
-		                                } else {
-		                                    echo date('d', $currentDayTimestamp).'<br/>';
-		                                    echo "<span>".date('M', $currentDayTimestamp).'</span>';
-		                                }
-		                                echo '</td>';
+// define attendance tables, if user is permit to view them
+if (isActionAccessible($guid, $connection2, '/modules/Attendance/attendance.php') && isset($_SESSION[$guid]["username"])) {
+
+    // proto attendance table with columns for both
+    // roll group and course class
+    $dailyAttendanceTable = DataTable::create(
+        '',
+        (new SimpleRenderer)
+            ->addClass('mini')
+            ->addClass('dailyAttendanceTable')
+    );
+    $dailyAttendanceTable->addColumn('group', __('Group'))
+        ->width('80px')
+        ->format(function ($row) use ($session) {
+            return Format::link(
+                $session->get('absoluteURL') . '/index.php?'.
+                    http_build_query(['q' => $row['groupQuery'], $row['rowID'] => $row[$row['rowID']]]),
+                $row['groupName']
+            );
+        });
+    $dailyAttendanceTable->addColumn('recent-history', __('Recent History'))
+        ->width('342px')
+        ->format(function ($row) {
+            $dayTable = "<table class='historyCalendarMini'>";
+
+            $l = sizeof($row['recentHistory']);
+            for ($i = 0; $i < $l; $i++) {
+                $dayTable .= '<tr>';
+                for ($j = 0; ($j < 10) && ($i + $j < $l); $j++) {
+                    // grouping 10 days as a row
+                    $day = $row['recentHistory'][$i + $j];
+                    $link = '';
+                    $content = '';
+
+                    // default link and content
+                    if (!empty($day['currentDate']) && !empty($day['currentDayTimestamp'])) {
+                        // link and date content of a cell
+                        $link = './index.php?' . http_build_query([
+                            'q' => '/modules/Attendance/attendance_take_byCourseClass.php',
```

Comment:
Hmm, this is also linking to `attendance_take_byCourseClass.php` for the Form Group/Roll Group, which results in an error 🤔 Should be `attendance_take_byRollGroup.php` for that case.